test/dist_autograd_test.py

```diff
@@ -1292,5 +1292,77 @@ def test_backwards_nested_python_udf(self):
             self.assertEqual(t1.grad, grads[t1])
             self.assertEqual(t2.grad, grads[t2])
 
+
+    _test_clean_context_backward_context_id = None
+
+    class MyBackwardFunc(Function):
+        @staticmethod
+        def forward(ctx, input):
+            return input
+
+        @staticmethod
+        @once_differentiable
+        def backward(ctx, input):
+            assert(DistAutogradTest._test_clean_context_backward_context_id is not None)
+
+            # Release the context to simulate error (use barrier before releasing context to ensure all nodes execute the backward function).
+            dist.barrier()
+            dist_autograd._release_context(DistAutogradTest._test_clean_context_backward_context_id)
```

Comment:
Is this blocking? It has to be to guarantee correctness, right?

Reply:
This is not blocking, it releases the local context and then just sends async RPCs to release other contexts. The method below `_all_contexts_cleaned_up` is blocking and ensures that contexts are cleaned up on all nodes.